### DISCUSSION TITLE
Keycloak metrics in penseive grafana

### DISCUSSION THREAD

**Discussion by shaik-jhani** [keycloak/keycloak#24916] [Q&A]
Hi Team, currently we are using the **Keycloak version as 21.1.1**. We used the **"keycloak-event-metrics-1.0.1-SNAPSHOT.jar"** to expose the metrics from keycloak to penseive grafana.

But in penseive we are not getting the below metrics which will help us to monitoring the realtime actions.

1.     keycloak_failed_login_attempts
2. 	keycloak_failed_authentication_attempts
3. 	keycloak_failed_authorization_attempts
4. 	keycloak_login_success_total
5. 	keycloak_login_failure_total
6. 	keycloak_authorization_success_total
7. 	keycloak_authorization_failure_total
8. 	keycloak_server_state

So, Can you please help to know do you have any keycloak version jar which will help us to get above metrics in penseive.
or
Do we have any alternate approach to get this metrices for keycloak in penseive ?

**Comment by sirishkumar** [comment]
With keycloak 21.0 there was a change in metric names. You can check metrics endpoint and their descriptions 

For example 
1. keycloak_login_success_total : # HELP keycloak_logins_total Total successful logins
2. keycloak_login_failure_total: # HELP keycloak_failed_login_attempts_total Total failed login attempts

**Reply by shaik-jhani** [reply]
Hi Sirish,

Thanks for the response. But, we are not able to get those metrics expect in the attachment. 

But, We need those metrics which is help us to monitor for more performance and error rates.
![Screenshot 2023-11-22 183941](https://github.com/keycloak/keycloak/assets/112946323/0e12cf0a-0c5c-4c09-acb9-74fc1a11d722)

**Reply by shaik-jhani** [reply]
Can you please help to suggest to get those additional metrics for Keycloak 21.1.1
